Source/Csla.Blazor/ApplicationContextManager.cs

```diff
@@ -0,0 +1,162 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="ApplicationContextManager.cs" company="Marimer LLC">
+//     Copyright (c) Marimer LLC. All rights reserved.
+//     Website: https://cslanet.com
+// </copyright>
+// <summary>Application context manager that uses HttpContextAccessor</summary>
+//-----------------------------------------------------------------------
+using Csla.Core;
+using Microsoft.AspNetCore.Components.Authorization;
+using System;
+using System.Security.Claims;
+using System.Security.Principal;
+using System.Threading.Tasks;
+
+namespace Csla.Blazor
+{
+  /// <summary>
+  /// Application context manager that uses HttpContextAccessor when 
+  /// resolving HttpContext to store context values.
+  /// </summary>
+  public class ApplicationContextManager : IContextManager, IDisposable
+  {
+    private ContextDictionary LocalContext { get; set; }
+    private ContextDictionary ClientContext { get; set; }
+    private IPrincipal CurrentPrincipal { get; set; }
+    private readonly ClaimsPrincipal UnauthenticatedPrincipal = new();
+    private bool disposedValue;
+
+    /// <summary>
+    /// Gets the current HttpContext instance.
+    /// </summary>
+    protected AuthenticationStateProvider AuthenticationStateProvider { get; private set; }
+
+    /// <summary>
+    /// Gets or sets a reference to the current ApplicationContext.
+    /// </summary>
+    public ApplicationContext ApplicationContext { get; set; }
+
+    /// <summary>
+    /// Creates an instance of the object, initializing it
+    /// with the required IServiceProvider.
+    /// </summary>
+    /// <param name="authenticationStateProvider">AuthenticationStateProvider service</param>
+    public ApplicationContextManager(AuthenticationStateProvider authenticationStateProvider)
+    {
+      AuthenticationStateProvider = authenticationStateProvider;
+      CurrentPrincipal = UnauthenticatedPrincipal;
+      AuthenticationStateProvider.AuthenticationStateChanged += AuthenticationStateProvider_AuthenticationStateChanged;
+      InitializeUser();
+    }
+
+    private void InitializeUser()
+    {
+      AuthenticationStateProvider.GetAuthenticationStateAsync().ContinueWith((task) =>
+      {
+        if (task.IsCompletedSuccessfully && task.Result != null)
+          CurrentPrincipal = task.Result.User;
+        else
+          CurrentPrincipal = UnauthenticatedPrincipal;
+      });
+    }
+
+    private void AuthenticationStateProvider_AuthenticationStateChanged(System.Threading.Tasks.Task<AuthenticationState> task)
+    {
+      CurrentPrincipal = task.Result.User;
```

Comment:
I think this probably should use the same technique of accessing the Result only after a continuation has run. I don't know for sure that the even would be raised before the result is available, but I think it's possible based upon the design of the API.